### DISCUSSION TITLE
modularization folders architecture

### DISCUSSION THREAD

**Discussion by amine2233** [pointfreeco/episode-code-samples#110] [Q&A] [answered]
tank you for your awesome work

i have a question about architecture all the folders and sub-folders, in modularization project.

what is the differences between this two approaches ?

first approach

we put each modules in separated module like bellow

```
|- Module-1
  |- Package.swift
  |- Source
    |- Module-1
  |- Tests
    |- Module-1-Tests
|- Module-2
  |- Package.swift
  |- Source
    |- Module-2
  |- Tests
    |- Module-3-Tests
|- Module-3
  |- Package.swift
  |- Source
    |- Module-3
  |- Tests
    |- Module-3-Tests
|- Module-4
  |- Package.swift
  |- Source
    |- Module-4
  |- Tests
    |- Module-4-Tests
```

we put each modules in one Root module like bellow

```
|- RootModule
  |- Package.swift
  |- Source
    |- Module-1
    |- Module-2
    |- Module-3
    |- Module-4
  |- Tests
    |- Module-1-Tests
    |- Module-3-Tests
    |- Module-3-Tests
    |- Module-4-Tests
```

Whats is Pro and Cons between this two approaches

Xcode issue ? when importing the modules ?
Conflict Dependencies if each module use same dependency example (Module1 & Module3) use same dependency (Alamofire) ?
Maintain ?
TimeBuild ?
Git ?

**Comment by stephencelis** [comment, marked as answer]
Hi @amine2233, we think that maintaining a single package with multiple targets (the second) is much easier to maintain than multiple packages with single targets (the first). We're not sure there's any benefit, in fact, to splitting a package out into several unless you want to ship the libraries of that package as a standalone repo, for example if you maintain an open source library. When it comes to modularizing an app's code base, we think a single package is fine, and a lot less work :smile:

**Reply by amine2233** [reply]
thanks @stephencelis for your response,
